pyopia/pipeline.py

```diff
@@ -68,12 +83,19 @@ class Pipeline():
 
         pyopia.io.show_h5_meta(datafile_hdf + '-STATS.h5')
 
+
     '''
 
     def __init__(self, steps):
-        self.steps = steps
-        self.common = self.steps['common']()
-        self.cl = self.steps['classifier']()
+        print('Initialising pipeline')
+        self.common = dict()
```

Comment:
Do you want this dict to contain both the data and the steps? If you don't need that to be the case then  I would not use the same dict. It just seems to add a kind of confusion for me.

If possible, have a "self.data" with the inputs/outputs of steps (see comment above, this could also be self.common if you prefer that word).
And here I would use a self.steps which would be the common['steps'] as you have it now.

Reply:
fixed :)